### ISSUE TITLE
Corrupt TS 3.4 declaration files in multiple packages

### ISSUE THREAD

**Issue by thefat32** [aws/aws-sdk-js-v3#3847]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

After installing, the package contains a corrupt typescript 3.4 declaration file.

For example when installing `node_modules\@aws-sdk\client-lex-runtime-v2` the following file is corrupted:

`node_modules\@aws-sdk\client-lex-runtime-v2\dist-types\ts3.4\models\models_0.d.ts`

This only affects projects using typescript version `<4` as `dist-types\ts3.4` is set to be used for versions lower to 4 in package.json files using `typesVersions` option

### SDK version number

@aws-sdk/client-lex-runtime-v2@3.142.0

### Which JavaScript Runtime is this issue in?

All

### Details of the browser/Node.js/ReactNative version

node: v16.14.2  - yarn: 1.22.15

### Reproduction Steps

Intall package in any project with `yarn install @aws-sdk/client-lex-runtime-v2`

### Observed Behavior

The declaration file is corrupt and breaks compilations that use imports contained in it

Example error:

```
message: `/root/amplify-js/node_modules/@aws-sdk/client-lex-runtime-v2/dist-types/ts3.4/commands/GetSessionCommand.d.ts (4,29): Module '"../models/models_0"' has no exported member 'GetSessionResponse'. Did you mean 'PutSessionResponse'?`
```

### Expected Behavior

The declaration file should be fine

### Possible Solution

Stop using `strip-comments` and search an alternative or stop stripping comments from declaration files at all

### Additional Information/Context

The declaration file is generated using `downlevel-dts` as you can see in [package.json](https://github.com/aws/aws-sdk-js-v3/blob/5ab6a9febd8d7dd53538884af7e071e082d4bd43/clients/client-lex-runtime-v2/package.json#L11)
 
The original declaration file is fine, located at:
`node_modules\@aws-sdk\client-lex-runtime-v2\dist-types\models\models_0.d.ts`
  
 Tried to downlevel the declaration file locally using same `downlevel-dts` [version](https://github.com/aws/aws-sdk-js-v3/blob/5ab6a9febd8d7dd53538884af7e071e082d4bd43/clients/client-lex-runtime-v2/package.json#L64) as in this project `0.7.0` and I got a correct file. So it does not seem to be a bug in `downlevel-dts`.
 
 Bug starts at `3.36.1` version, older versions contains not corrupted declarations file
 
 EDIT: Found the problem:
 
 `strip-comments` used in this [script ](https://github.com/aws/aws-sdk-js-v3/tree/main/scripts/downlevel-dts/downlevelWorkspace.mjs) is breaking the file, could reproduce the bug locally in my pc
 
Couldn't find any buildspec in this repo, so I think you are managing deploy configurations in a private repo, the script using  `strip-comments` shoould be called in the buildspec and probably it's generating corrupt files in other packages too

**Comment by carvantes** [comment]
Is there an ETA to fix this?
I'm having the same issue with `@aws-sdk/client-elastic-load-balancing-v2/dist-types/ts3.4/models/models_0.d.ts`

I ended up upgrading to typescript 4.x to circumvent this issue

**Comment by trivikr** [comment]
I created a bug report upstream providing an example when strip-comments strips source code https://github.com/jonschlinkert/strip-comments/issues/75

**Comment by thefat32** [comment]
Nice solution! I don't think we'll get a fix from strip-comments developer, the package seems to be abandoned.

**Comment by trivikr** [comment]
This issue was fixed in `v3.170.0`

```console
$ npm install --exact @aws-sdk/client-lex-runtime-v2@3.170.0

$ tail node_modules/@aws-sdk/client-lex-runtime-v2/dist-types/ts3.4/models/models_0.d.ts 
 */
export declare const RecognizeTextResponseFilterSensitiveLog: (obj: RecognizeTextResponse) => any;
/**
 * @internal
 */
export declare const StartConversationResponseEventStreamFilterSensitiveLog: (obj: StartConversationResponseEventStream) => any;
/**
 * @internal
 */
export declare const StartConversationResponseFilterSensitiveLog: (obj: StartConversationResponse) => any;
```
